pkg/apis/azure/v1alpha1/register.go

```diff
@@ -35,9 +35,12 @@ const (
 	// Version is the version of the API group
 	Version = "v1alpha1"
 	// APIVersion is the full version of this API group
-	APIVersion             = Group + "/" + Version
-	ProviderKind           = "provider"
-	ProviderKindAPIVersion = ProviderKind + "." + APIVersion
+	APIVersion        = Group + "/" + Version
+	ProviderKind      = "provider"
+	ResourceGroupKind = "resourcegroup"
```

Comment:
@jbw976 and I talked about this a little bit. My thought would be that resources that do not have a an active compute action (i.e. they aren't "running", they are simply "existing") should be included here.

Reply:
Another point here is that resource groups are a core property of basically every resource in the Azure Resource Manager based API: https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-overview
It felt like a good fit at the core of the Azure API group because it's consumed by all other Azure types and it felt a bit heavy to make its own package just for it.

Reply:
That all makes sense to me. 👍